draft-plugin/server.go

```diff
@@ -0,0 +1,200 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/rest"
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	v1 "k8s.io/test-infra/prow/client/clientset/versioned/typed/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pjutil"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins/trigger"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func helpProvider(_ []config.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: `The darft plugin is used for trigger job when darft PR or convert PR to draft. Useful when you want to trigger job when PR is in draft`,
+	}
+	return pluginHelp, nil
+}
+
+type server struct {
+	tokenGenerator func() []byte
+	botUser        *github.UserData
+	email          string
+	gc             git.ClientFactory
+	ghc            github.Client
+	log            *logrus.Entry
+	repos          []github.Repo
+	prowconfig     string
+	prowjob        string
+	regex          string
+	ns             string
+}
+
+type githubClient interface {
+	GetRef(org, repo, ref string) (string, error)
+}
+
+type prowJobClient interface {
+	Create(context.Context, *prowapi.ProwJob, metav1.CreateOptions) (*prowapi.ProwJob, error)
+	List(ctx context.Context, opts metav1.ListOptions) (*prowapi.ProwJobList, error)
+	Update(context.Context, *prowapi.ProwJob, metav1.UpdateOptions) (*prowapi.ProwJob, error)
+}
+
+func (s *server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	logrus.Info("inside http server")
+	eventType, eventGUID, payload, ok, _ := github.ValidateWebhook(w, r, s.tokenGenerator)
+	if !ok {
+		return
+	}
+	logrus.Info(w, "Event received. Have a nice day.")
+	if err := s.handleEvent(eventType, eventGUID, payload); err != nil {
+		logrus.WithError(err).Error("Error parsing event.")
+	}
+}
+
+func (s *server) handleEvent(eventType, eventGUID string, payload []byte) error {
+	l := logrus.WithFields(
+		logrus.Fields{
+			"event-type":     eventType,
+			github.EventGUID: eventGUID,
+		},
+	)
+	_ = l
+	var c *config.Config
+	var p = getProwClient()
+	c, err := config.Load(s.prowconfig, s.prowjob, nil, "")
+	if err != nil {
+		logrus.Info(err)
+	}
+	var pr github.PullRequestEvent
+	if err := json.Unmarshal(payload, &pr); err != nil {
+		return err
+	}
+	org, repo, author, ref := orgRepoAuthor(pr.PullRequest)
+	baseSHA := ""
```

Comment:
```suggestion
	var baseSHA string
```
This should be at start of the function as the scope of this variable is entire function